### ISSUE TITLE
#2020 introduced a subtitle selection bug

### ISSUE THREAD

**Issue by dathbe** [jellyfin-archive/jellyfin-roku-legacy#2049]
## Software Versions

- Jellyfin Server Version:  10.10.1
- Roku Client Version:  2.2.2 / 2.2.3

## Describe the bug
After the inclusion of #2020 in v2.2.2, a new bug has been introduced.  Specifically, if Roku is set to NOT display subtitles globally, and a video with default subtitles is played, the subtitles will not appear (as I think is expected?), but when you go to turn on the subtitles, the default subtitles already show as selected.  That requires setting the subtitle to off, exiting the subtitle screen, going back into the subtitle screen, setting the subtitle to on, and then going back out of the subtitle screen before the subtitles can be displayed.

## How To Reproduce
1. Turn off subtitles globally for Roku
2. Play a video with a "default" subtitle in Jellyfin-Roku
3. Attempt to turn on subtitles in Jellyfin-Roku menu
4. Bug occurs

## Expected behavior
I see four possible solutions:

1) Make sure that "None" is selected for all videos and all subtitle types when Roku global subtitles are turned off.
2) Make sure that "default" subtitles are displayed on a video even though global subtitles are turned off (this is what I understand happens for "forced" subtitles after #2020).
3) The presence of a "default" subtitle on a video will force the global subtitles setting to be on.  This is the behavior for <v2.2.2.  I liked it, but I'll admit it's probably not what most people would expect.
4) Have Jellyfin remember the user's setting for subtitles.  If a default subtitle was on for the last video, it will be on for the next video, even after restarting the app.  If it was off, it will be off.

I'd prefer the 4th option.  I really hate the fact that subtitles are mostly global on Roku.  I want subtitles on scripted shows and movies, but I do not want them on live tv like sports or news.  These are typically separate apps, so an app-by-app setting would be a better way to handle subtitles.  But that might be overly complicated. 

I'll begrudgingly admit that the 1st option is probably the next best option based on what most people would expect to happen with "default" subtitles...even though I'd prefer the 2nd option. 

## Logs
n/a

## Screenshots
![Image](https://github.com/user-attachments/assets/8fce2292-b33b-4e90-85ae-6d1a794056e6)

## Connection Information

- Is server local or remote?  both
- Is server connection HTTP or HTTPS? https

## Additional context
n/a.  Happy to answer any questions.

**Comment by jimdogx** [comment]
I wasn't able to reproduce this, but looking at the code, there was a one line logic error that would probably contribute to what you are seeing.

Fixed by https://github.com/jellyfin/jellyfin-roku/pull/2050

**Comment by dathbe** [comment]
I'm happy to test the fix if someone can create the beta zip file for #2050

**Comment by jimdogx** [comment]
> I'm happy to test the fix if someone can create the beta zip file for [#2050](https://github.com/jellyfin/jellyfin-roku/pull/2050)

The Zips get created automatically 😄 

https://github.com/jellyfin/jellyfin-roku/actions/runs/11864274298

**Comment by dathbe** [comment]
I've never been able to figure out how to navigate from a PR to those files.  Is there a trick?

**Comment by dathbe** [comment]
#2050 does seem to fix the problem I was seeing.  Should I close this issue now, or wait for the PR to get merged into the main branch?

**Comment by jimdogx** [comment]
> [#2050](https://github.com/jellyfin/jellyfin-roku/pull/2050) does seem to fix the problem I was seeing. Should I close this issue now, or wait for the PR to get merged into the main branch?

It will get closed when the PR is merged 👍

**Comment by jimdogx** [comment]
> I've never been able to figure out how to navigate from a PR to those files. Is there a trick?

Just click on Actions (instead of Issues or Pull Requests)... then make sure you're looking at "All Workflows" (upper left).... then scroll until you find the title of the PR in question that you want to test 👍

**Comment by cewert** [comment]
@dathbe click the green checkmark (or red X) next to the commit you want to test, then click "details" on build-dev (or build-prod if it's listed) OR go to the bottom of any PR and click "show all checks" then click details for build-dev. Once you're on the details screen, click "Summary" in the top left. The summary page will show a zip file at the bottom you can download.

![Image](https://github.com/user-attachments/assets/5332a8ac-e950-45a4-af39-607cba66daf6)

![Image](https://github.com/user-attachments/assets/259d31d0-98b4-46e8-aa9c-aa5a26a4f30f)

![Image](https://github.com/user-attachments/assets/fac7b00b-b6ec-43a7-b679-2343fbaf5e66)

**Comment by jimdogx** [comment]
Fixed by #2050
